src/gc.rs

```diff
@@ -23,6 +24,11 @@ type Address = usize;
 
 type Word = usize;
 
+/// Use this type when we do not care about the contents of a Gc. We choose `u8`
+/// because it maps similarly to how C / C++ use char when dealing with raw
+/// bytes.
+pub(crate) type Placeholder = u8;
```

Comment:
This might be slightly dangerous as `type` in Rust doesn't create a new type, but just an alias. That means that a user who has a `Gc<u8>` struct will find that they are exactly equivalent to `Gc<Placeholder>`. One way around this might be to make this a newtype of sorts, maybe just:

```rust
struct Placeholder {
    byte: u8
}
```

Provided that struct can't escape the crate, there's then no chance of a clash.

Reply:
Alternatively, maybe just embrace the `u8`ness and delete the `type` alias?

Reply:
I will play with the newtype idea of `Placeholder` first, as I find it nicely self-documents the distinction between `Gc<Everything>` and some actual `Gc<u8>`.

Reply:
Actually, I was wondering if it should be a zero-sized type, because then you wouldn't give the illusion of sized-ness to the outside world. In other words, just `struct Placeholder;` would work I think?

Reply:
A lot of the pointer arithmetic code relies on the `Placeholder` type being 1 byte sized. For simplicity, I've decided to keep it the same size. Fixed: 77c9918badbba02c7dfc27a709e27e874f4041b0

Reply:
> A lot of the pointer arithmetic code relies on the Placeholder type being 1 byte sized.

I hadn't realised that. Then we should probably find a better name for `Placeholder` that makes that clearer. `OpaqueU8` or similar perhaps?

Reply:
Yeah I can't really think of a better name than that. Try now